### PR TITLE
fix(core): mark all projects as affected when npm package is deleted

### PR DESCRIPTION
**Pull request by jaysoo**
_[Please make sure you have read the submission guidelines before posting an PR](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#submit-pr)_

> _Please make sure that your commit message follows our format._

> Example: `fix(nx): must begin with lowercase`

### CODE REVIEW

File: packages/workspace/src/core/affected-project-graph/locators/npm-packages.ts [hunk 1]
```diff
@@ -1,30 +1,37 @@
...
+
+  for (const c of changes) {
+    if (
+      isJsonChange(c) &&
+      (c.path[0] === 'dependencies' || c.path[0] === 'devDependencies')
+    ) {
+      // A package was deleted so mark all workspace projects as touched
+      // since we don't know which one used to depend on it.
+      if (c.type === DiffType.Deleted) {
+        touched = Object.keys(nxJson.projects);
```
Comment (FrozenPandaz) [inline packages/workspace/src/core/affected-project-graph/locators/npm-packages.ts]:
```suggestion
      touched = Object.keys({
        ...(packageJson.dependencies || {}),
        ...(packageJson.devDependencies || {})
      });
```

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Comment by vsavkin** [comment]
I'm not sure if this is the right solution. Consider the following:

* Project A depends on NodeModule B and nothing else.
* We removed B without updating A
* A doesn't depend on any other package, so them being touched does nothing to A, but A is broken.

Right?

**Comment by vsavkin** [comment]
@jaysoo can you explain the rationale for this fix?

Given the example deleting something should probably result in all projects (not packages) marked as affected.

**Comment by jaysoo** [comment]
@vsavkin Yeah I spoke to Jason about it a while back and it should be all workspace projects -- just hadn't had time to update PR yet.

**Comment by jaysoo** [comment]
Closed and re-opened as https://github.com/nrwl/nx/pull/2508 (Circle wasn't picking up the unit test fix for some reason).

**Pull request closed without merging**